src/pyfaf/opsys/centos.py

```diff
@@ -192,10 +192,17 @@ def get_releases(self):
     def get_components(self, release):
         urls = [repo.replace("$releasever", release)
                 for repo in [self.base_repo_url, self.updates_repo_url]]
-
-        yum = Yum(self.name, *urls)
-        components = list(set(pkg["name"]
-                              for pkg in yum.list_packages(["src"])))
+        components = []
+        if "dnf" in repo_types:
+            from pyfaf.repos.dnf import Dnf
+            dnf = Dnf(self.name, *urls)
+            components.extend(list(set(pkg["name"]
+                                       for pkg in dnf.list_packages(["src"]))))
+        if "yum" in repo_types:
```

Comment:
Should not this be `elif`? If both, dnf and you are present, then there the packages will be duplicated. Also maybe `else` branch would be nice, which would raise FafError("No repo type available")